pkg/vcs/vcs.go

```diff
@@ -72,6 +72,9 @@ type Repo interface {
 
 	// CommitExists check for the commit presence in local checkout.
 	CommitExists(commit string) (bool, error)
+
+	// ArchiveHeadCommit is used to store commits elsewhere.
+	ArchiveHeadCommit(repo, commit string) error
```

Comment:
We are pushing commit `commit`, not the HEAD commit, so the method name shouldn't have "Head".
Also I would call it Push rather than Archive. The use we have now is for archiving, but effectively it's just push can be used for other things as well.